app/observers/cache_sweeper.rb

```diff
@@ -0,0 +1,10 @@
+class CacheSweeper < ActionController::Caching::Sweeper
+  observe Book, BookPresentation, Gathering
+
+  def after_commit(_record)
+    return unless @controller
```

Comment:
Could you clarify the purpose of this line?

Reply:
There can be a few cases where the @controller is nil. When that happens, we can't use URL helpers and the program crashes so this is a early exit.